hpOneView/resources/settings/firmware_bundles.py

```diff
@@ -55,8 +56,16 @@ def upload(self, file_path, timeout=-1):
             file_path: Full path to firmware.
             timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
                 in OneView; it just stops waiting for its completion.
+            compsig_path: Full path to the signature of the bundle. If omitted OneView might complain abouto missing signature.
 
         Returns:
-          dict: Information about the updated firmware bundle.
+          (dict, dict): Information about the updated firmware bundle and potential compsig signature.
         """
-        return self._client.upload(file_path, timeout=timeout)
+        bundle_res = self._bundle_client.upload(file_path, timeout=timeout)
+
+        if compsig_path is None or self._connection.apiVersion < 1000:
```

Comment:
Existing code doesn't seem to check the API version. I thought it's useful to check what's supported before attempting it, so that the same codebase can work with multiple versions.

There should probably be a warning to the user when they try to upload a compsig while using an older API version. But how do you do warnings in this project?